atlasdb-client/src/main/java/com/palantir/atlasdb/logging/SafeLoggableDataUtils.java

```diff
@@ -41,8 +46,13 @@ public static SafeLoggableData fromTableMetadata(Map<TableReference, byte[]> tab
 
         tableRefToMetadata.forEach(
                 (ref, metadataBytes) -> {
-                    TableMetadata tableMetadata = TableMetadata.BYTES_HYDRATOR.hydrateFromBytes(metadataBytes);
-                    addLoggableNamesToBuilder(builder, ref, tableMetadata);
+                    try {
+                        TableMetadata tableMetadata = TableMetadata.BYTES_HYDRATOR.hydrateFromBytes(metadataBytes);
+                        addLoggableNamesToBuilder(builder, ref, tableMetadata);
+                    } catch (Exception e) {
+                        log.warn("Exception thrown hydrating table metadata for table {}. {}",
+                                UnsafeArg.of("tableName", ref), UnsafeArg.of("exception", e));
```

Comment:
Usually you can just leave the exception on at the end without needing to wrap it in an Arg. This way you get the stack trace (which is considered safe), though the (possibly unsafe) message is redacted.